### PR TITLE
Make REST API authentication configurable

### PR DESCRIPTION
**Pull request by afalhambra-hivemq**
https://hivemq.kanbanize.com/ctrl_board/22/cards/18607/details/

### CODE REVIEW

**Review by mhofsche — APPROVED** [review]

File: tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/util/K8sUtil.java [hunk 1]
```diff
@@ -222,4 +225,15 @@ public static void waitForHiveMQPlatformStateRunning(
         hivemqCustomResource.waitUntilCondition(getHiveMQPlatformStatus("RUNNING"), 5, TimeUnit.MINUTES);
         assertThat(hivemqCustomResource.get().get("status").toString()).contains("RUNNING");
     }
+
+    /**
+     * Waits for all the pods to be deleted in the given namespace
+     * @param client    the Kubernetes client to use
+     * @param namespace the namespace to wait to check for all the pods to be removed
+     */
+    public static void waitForNoPodsDeletedInNamespace(
```
Comment (Donnerbart) [inline tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/util/K8sUtil.java:234]:
```suggestion
    public static void waitForAllPodsDeletedInNamespace(
```

File: tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmRestApiIT.java [hunk 2]
```diff
@@ -41,4 +100,52 @@ void platformChart_whenRestApiEnabled_thenCallsEndpoint() throws Exception {
...
+                    .get(new URL(baseRestApiEndpoint + "/api/v1/management/backups"))
+                    .then()
+                    .statusCode(HttpStatus.SC_FORBIDDEN);
+        }
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private static @NotNull String createBasicAuthHeader(
+            final @NotNull String username, final @NotNull String password) {
+        final String s = username + ":" + password;
```
Comment (Donnerbart) [inline tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmRestApiIT.java:148]:
```suggestion
        final var s = username + ":" + password;
```

File: tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmRestApiIT.java [hunk 3]
```diff
@@ -41,4 +100,52 @@ void platformChart_whenRestApiEnabled_thenCallsEndpoint() throws Exception {
...
+        return "Basic " + Base64.getEncoder().encodeToString(s.getBytes(StandardCharsets.UTF_8));
```
Comment (Donnerbart) [inline tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmRestApiIT.java:149]:
```suggestion
        final var credentials = username + ":" + password;
        return "Basic " + Base64.getEncoder().encodeToString(credentials.getBytes(StandardCharsets.UTF_8));
```

File: tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmRestApiIT.java [hunk 4]
```diff
@@ -41,4 +100,52 @@ void platformChart_whenRestApiEnabled_thenCallsEndpoint() throws Exception {
...
+        K8sUtil.waitForHiveMQPlatformStateRunning(client, NAMESPACE, PLATFORM_RELEASE_NAME);
+
+        // forward the port from the service
+        try (final var forwarded = K8sUtil.getPortForward(client,
+                NAMESPACE,
+                REST_API_SERVICE_NAME,
+                REST_API_SERVICE_PORT)) {
+            final var baseRestApiEndpoint = "http://localhost:" + forwarded.getLocalPort();
+
+            given().header("Authorization", createBasicAuthHeader("test-user", "test-password"))
```
Comment (Donnerbart) [inline tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmRestApiIT.java:125]: This looks very familiar with the ESE tests. I like it very much that you re-used the test idea/structure/code.

**Review by Donnerbart — APPROVED** [review]
Just some nitpicks.

As always, great new functionality and good tests!

File: tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmRestApiIT.java (hunk 4 above)
Comment (afalhambra-hivemq) [reply tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmRestApiIT.java:125]: Yeah, it's stolen from the ones we have there already :) - though I was kind of in doubt whether we really need to check that functionality here, but I guess there's no harm in double checking.

File: tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/util/K8sUtil.java (hunk 1 above)
Comment (afalhambra-hivemq) [reply tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/util/K8sUtil.java]: thanks! didn't realize what a weird name I used for that! :)

**Pull request merged**